webgoat-lessons/path-traversal/src/main/java/org/owasp/webgoat/path_traversal/ProfileUploadRemoveUserInput.java

```diff
@@ -0,0 +1,29 @@
+package org.owasp.webgoat.path_traversal;
+
+import org.owasp.webgoat.assignments.AssignmentHints;
+import org.owasp.webgoat.assignments.AttackResult;
+import org.owasp.webgoat.session.WebSession;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import static org.springframework.http.MediaType.ALL_VALUE;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@RestController
+@AssignmentHints({"path-traversal-profile-remove-user-input.hint1", "path-traversal-profile-remove-user-input.hint2", "path-traversal-profile-remove-user-input.hint3"})
+public class ProfileUploadRemoveUserInput extends ProfileUploadBase {
+
+    public ProfileUploadRemoveUserInput(@Value("${webgoat.server.directory}") String webGoatHomeDirectory, WebSession webSession) {
+        super(webGoatHomeDirectory, webSession);
+    }
+
+    @PostMapping(value = "/PathTraversal/profile-upload-remove-user-input", consumes = ALL_VALUE, produces = APPLICATION_JSON_VALUE)
```

Comment:
This does not match the HTML form. It results in status 400. The integration test does send the correct things, but from the HTML form different and incorrect data is sent.

HTTP/1.1 400 Bad Request
Connection: close
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
Content-Type: application/json
Date: Wed, 04 Mar 2020 18:21:02 GMT

{
  "timestamp" : "2020-03-04T18:21:02.308+0000",
  "status" : 400,
  "error" : "Bad Request",
  "message" : "Required request part 'uploadedFileRetrieval' is not present",
  "trace" : "org.springframework.web.multipart.support.MissingServletRequestPartException: Required request part 'uploadedFileRetrieval' is not

Reply:
Fixed wrong post through JavaScript